File: Lexer/LeftParenToken.cpp
#include "token.h"
#include <string>
using namespace std;

class LeftParen : public Token {
	bool Token::equals()
	{

	}

	int Token::hashCode()
	{
		return 3;
	}

	string Token::toString()
	{
		return "(";
	}
};

File: src/main/java/parser/LeftParenToken.java
package parser;

public class LeftParenToken implements Token{
	public boolean equals(final Object other) {
		return other instanceof LeftParenToken;
	}
	
	public int hashCode() {
		return 8;
	}
	
	public String toString() {
		return "LeftParenToken";
	}

}
